macOS App with Network Extension Implementation (Mac AppStore only)

yaroslavyaroslav opened the pull request:
Closes: #442, #455

yaroslavyaroslav committed:
There's a common issue of building iOS apps on Qt 6.8 because of new introduced ffmpeg dependency in multimedia Qt package
ref: https://community.esri.com/t5/qt-maps-sdk-questions/build-failure-on-ios-with-qt-6-8/m-p/1548701#M5339

yaroslavyaroslav committed:
Various entitlements

yaroslavyaroslav committed:
Resources changes

yaroslavyaroslav committed:
Source code changes

yaroslavyaroslav committed:
Ci-cd config update

yaroslavyaroslav committed:
Cmake related changes

yaroslavyaroslav committed:
Submodules updated

yaroslavyaroslav committed:
Remove me

yaroslavyaroslav committed:
QtWidget exclusion omitted

yaroslavyaroslav committed:
Distribution errors fixed

yaroslavyaroslav committed:
Merge branch 'dev' into feature/macos-network-extension-bak

yaroslavyaroslav committed:
Outdated files deleted

yaroslavyaroslav committed:
macos_ne cmake fixed

yaroslavyaroslav committed:
fix: update provisioning profile specifiers for macOS network extension

yaroslavyaroslav committed:
fix: update provisioning profile specifiers and code sign flags for macOS build

yaroslavyaroslav committed:
Revert me
(temporary 3rd-build commit pointer)

pincetgore commented:
Hello @yaroslavyaroslav is it possible to bump app version too in your PR?

yaroslavyaroslav commented:
@pincetgore It's still on a long way to be merged, so currently I'm planning to update version when it'll about to merge. 

However, if there's a reason to do it immediately please feel free to share it.

pincetgore commented:
> @pincetgore It's still on a long way to be merged, so currently I'm planning to update version when it'll about to merge.
> 
> However, if there's a reason to do it immediately please feel free to share it.

There is no reason to bump version before your PR is merged. 
OK, gonna continue to wait patiently for the day 1 for Mac users.

lunardunno commented:
[Deploy workflow #4287](https://github.com/amnezia-vpn/amnezia-client/actions/runs/13993919919) has been going on for 3.5 hours now.

yaroslavyaroslav committed:
fix: Welcome screen fix

yaroslavyaroslav committed:
fix: ci/cd hanging forever fix

yaroslavyaroslav committed:
fix: Fixed error popup on macos on file save

yaroslavyaroslav committed:
Merge remote-tracking branch 'origin/dev' into feature/macos-network-extension-bak

yaroslavyaroslav committed:
refactor: rename networkextension target to AmneziaVPNNetworkExtension in macos build configuration

yaroslavyaroslav committed:
Merge branch 'dev' into feature/macos-network-extension-bak

pincetgore commented:
Please don't be dead 🤞

yaroslavyaroslav commented:
Fear no more, it was just a vacation 💁🏻‍♂️

yaroslavyaroslav committed:
feat: add autostart support for Mac App Store builds on macOS

Fixes: QA-8

yaroslavyaroslav committed:
feat: add debug logging to Autostart functionality on macOS

yaroslavyaroslav committed:
Revert "feat: add autostart support for Mac App Store builds on macOS"

This reverts commit 3bd25656fb4986d01e5bd6dd265f7279a73bd2a8.

yaroslavyaroslav committed:
feat: add platform-specific close window behavior for macOS App Store build with Network Extension

Closes: QA-12

yaroslavyaroslav committed:
When the application starts with "Start minimized" enabled on macOS (especially the
sandboxed App-Store build compiled with MACOS_NE), fully hiding the window prevents it
from being restored by clicking the Dock icon. The proper behaviour is to start the
window in the *minimized* state instead. That way the window is still part of the
window list and the system automatically brings it back when the user clicks the Dock
icon, replicating the native experience.

On the other platforms we keep the old behaviour (hide the window completely and rely
on the tray icon), therefore we switch at runtime by checking the current OS.

Closes: QA-7

Closes: QA-8

dmitrytheg commented:
Great to see that this is being developed! As fas as I understand this will allow adding amnezia vpn to mdm configs!

yaroslavyaroslav committed:
Revert "When the application starts with "Start minimized" enabled on macOS (especially the"

This reverts commit 7b0d17987cdfdbc4cedc3822bf3fd2e4973da452.

yaroslavyaroslav committed:
feat: MACOS_NE systray menu support

yaroslavyaroslav committed:
feat: add macOS notification handler and install event filter on main window

yaroslavyaroslav committed:
feat: implement custom close behavior for Amnezia application on different platforms

yaroslavyaroslav committed:
Merge branch 'dev' into feature/macos-network-extension-bak

yaroslavyaroslav committed:
fix: update provisioning profile specifiers for macos builds

yaroslavyaroslav committed:
Merge remote-tracking branch 'origin/dev' into feature/macos-network-extension-bak

yaroslavyaroslav committed:
fix: Fatal error in logs

CLI-216

yaroslavyaroslav committed:
fix: disabled unavailable on macos ne service logs

yaroslavyaroslav committed:
fix: dock icon now hides only when window is closed; menubar icon shows always

Initial state of the docker icon to be presented follows "Start minimized" setting in app settings.

yaroslavyaroslav committed:
temp-fix: temporary disable all OpenVPN options of VPN on MACOS_NE since it's not working yet.

yaroslavyaroslav committed:
fix: build script updated

yaroslavyaroslav committed:
feat: add macOS NE build workflow to GitHub Actions

yaroslavyaroslav committed:
fix: Not working Auto start toggle is hidden

yaroslavyaroslav committed:
fix: Log spamming during xray connection fixed

yaroslavyaroslav committed:
Merge branch 'dev' into feature/macos-network-extension-bak

yaroslavyaroslav committed:
3rd-prebuild points to commit that stores macos_ne universal binaries.

yaroslavyaroslav committed:
fix: missing native dependency on linking stage fixed

Nethius committed:
chore: update link to submodule

yaroslavyaroslav committed:
Merge branch 'dev' into feature/macos-network-extension-bak
